### DISCUSSION TITLE
MDM profile

### DISCUSSION THREAD

**Discussion by ghost** [usnistgov/macos_security#288] [Q&A] [answered]
Hi... I have this application block in my MDM profile but it doesn't seem to get enabled - what am I doing wrong?

.... 
		        <key>allowFingerprintForUnlock</key>
			<false/>
			<key>allowUSBRestrictedMode</key>
			<true/>
			<key>familyControlsEnabled</key>
			<true/>
			<key>pathBlackList</key>
			<array>
				<string>/System/Applications/News.app</string>
				<string>/System/Applications/Siri.app</string>
				<string>/System/Applications/Stocks.app</string>
				<string>/System/Applications/Freeform.app</string>
				<string>/System/Applications/Home.app</string>
				<string>/System/Applications/Time\ Machine.app</string>
				<string>/System/Applications/Voice\ Memos.app</string>
				<string>/System/Applications/Shortcuts.app</string>
			<string>/System/Applications/Utilities/Bluetooth\ File\Exchange.app</string>
			</array>
		</dict>
	</array>
	<key>PayloadDescription</key>
	<string>This payload disables iCloudPrivateRelay. Version 1</string>

I have seen, in the past,  some of Apple's apps talking on their own/bootup with aApple and I want to block them... plus with the class action lawsuit by the investor who found apple monitoring his investment strategy via Stocks or something on his system, I'd like to block many of their apps.

**Comment by brodjieski** [comment]
Which payloadtype are you applying these settings to?  Also, note that Apple deprecated the `pathBlackList` key back in macOS 10.15... so your milage may vary.

**Reply by ghost** [reply]
Ahhh. This was taken from the Ventura guidance. It is part of the application access payload.

**Reply by ghost** [reply]
I also have a key for disabling Private Relay and they didn't work either, even though it shows in Profile as a restriction. But sure enough Private Relay turned on for the Wi-Fi network I connected to and the option is not greyed out. But these application restrictions don't even show up. I hope that makes sense.

**Reply by brodjieski** [reply, marked as answer]
For the application restrictions, the `payloadtype` is com.apple.applicationaccess.new, and not the same com.apple.applicationaccess that the other restrictions are set.  Can you verify that you are using this in your profile?

**Reply by brodjieski** [reply]
@vaughnhart If these still aren't working for you after verify the payloadtype, you may need to look into third-party options.

**Comment by ghost** [comment]
<img width="729" alt="Screenshot 2023-08-17 at 5 01 16 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/30536628-c021-45cc-9fff-4758cfe42eda">
<img width="463" alt="Screenshot 2023-08-17 at 4 58 19 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/0b5259df-4e99-4528-b80f-99611b2e85b3">

As you can see Private Relay turned on with this connection... though it has never been on before.

**Reply by brodjieski** [reply]
You'll actually find the iCloud Private Relay settings under the iCloud System Settings panel, rather than the wireless connection settings.
<img width="827" alt="Screenshot 2023-08-17 at 6 48 36 AM" src="https://github.com/usnistgov/macos_security/assets/7014883/c445514c-c8ae-4379-bf7f-47e78aff53e4">

Can you check there to see if it's correctly disabled?

**Reply by ghost** [reply]
Yeah. I don't have that service but I am noticing on my system it's always showing up under work even after I turn it off. Also the real issue is the MDM profile isn't disabling that switch or any switch. Plus apple is nesting their QUIC protocol under their storage service. And I can't block any apps. I'm not trying to ignore you. I'm looking at the other class action lawsuit against Google, who developed QUIC, for reading your stuff in Incognito mode, and wondering what's really being blocked anymore by profiles. 

I have a case open with Cisco about the iCloud nesting. But when DoQ (DNS over QUIC) goes live - how can you stop it? Read the ITEF - they say evasion is possible, plain as day.

**Reply by brodjieski** [reply]
If you don't have that service, then how is it showing up on your system?  The disabling of private relay does work, as does the application restrictions (to some extent).

**Reply by jmahlman** [reply]
~~Limit IP address tracking is not Private relay. Private relay is separate and requires an Apple ID. If you're not using an Apple ID, or blocking the ability to log into an Apple ID, the private relay setting isn't going to have an effect.~~

Edit: Apparently, Apple considers this part of Private relay [[Source](https://support.apple.com/en-us/HT213548)]. But as I mention below, they give no way to control it on the device. You need to open feedback on this, it's not a missing control on the project's part.

Also as a note, setting it to false doesn't mean it turns it off, it simply disables the ability to change the setting.

**Reply by ghost** [reply]
@brodjieski @brodjieski 

A) I am using my AppleID (in a limited fashion) and I do not have iCloud+
B) there seems to be three different places for Apple's "Limit IP Address/Private Relay". The one in Wi-Fi, the one in AppleID and the one in Safari, all which seem to be indecently enabled from one another. 
C) I have seen and am seeing (without the iCloud+ service) traffic being logged (in Cisco Umbrella) to the mask.iCloud.con and mask-h2.iCloud.com service. 
D) there is a Reddit (not as reputable) that someone else has seen the same thing. 
D)I am completely able to toggle that Limit IP address switch on and off with the setting. And it turns itself on, even when I don't want it. ![image](https://github.com/usnistgov/macos_security/assets/102542366/fc803aa8-a7c1-433d-9066-16f2f1ee307e)

**Reply by jmahlman** [reply]
Limit IP address tracking is **not** controlled by limiting Private Relay. At present, Apple gives no way to control it on the device. You need to open feedback.

If you want to control it at a network level, [Apple has an article for that](https://developer.apple.com/support/prepare-your-network-for-icloud-private-relay).

**Reply by ghost** [reply]
Thank you. I saw the same article... but need to take a deeper look at it an the IETFs for QUIC. 

Do you have any suggestions for application control which this Q&A is really about? It a part of the Ventura guidance but it doesn't work. My thought is control the binaries and you control the world. 

😭😄

**Comment by jmahlman** [comment]
There is a note in the the application blocking controls:

> Apple has deprecated the use of [restriction controls](https://github.com/apple/device-management/blob/eb51fb0cb9626cac4717858556912c257a734ce0/mdm/profiles/com.apple.applicationaccess.new.yaml#L67-L70[application), using these controls may not work as expected. Third party software may be required to fulfill the compliance requirements.

If they're not working, you may need to look into a third party solution.

**Comment by ghost** [comment]
OK... that answers everything. Man... I miss GPO. You can almost do everything on Windows with GPO.

**Comment by ghost** [comment]
@brodjieski I am not using the new application access restriction payload. It's the old one. That's very important.  Thank you!

**Comment by ghost** [comment]
Thank you guys... I jerry-rigged something together... and it seems to work but there is a bypass.  And paths with a space seem to not be working. 
<img width="594" alt="Screenshot 2023-08-17 at 11 03 58 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/d7b805dc-58d4-47ee-a7bd-c79dc373c612">

**Reply by ghost** [reply]
Maybe sorted... do you see the discrepancy in Voice Memos... not pictured but it looks different in finder.  All I need to find now is the binary for Private Relay.  But this might all break in Sonoma. 

<img width="494" alt="Screenshot 2023-08-17 at 11 08 07 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/68e283bc-ac7b-4dcf-9102-7297582ffd96">
<img width="1220" alt="Screenshot 2023-08-17 at 11 11 11 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/e3e1f5c0-2d67-4460-9313-8a60c6ee931c">

**Reply by ghost** [reply]
Thank you. If I didn't say it before. Thanks!

-Vaughn

**Reply by ghost** [reply]
<img width="434" alt="Screenshot 2023-08-18 at 8 08 02 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/7fd734f3-b32b-4221-acc1-7663ec069053">
<img width="823" alt="Screenshot 2023-08-18 at 8 07 52 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/2663c565-3271-4132-90b4-7f7498ff5324">
interesting things I found when Limit IP Address Tracking enabled on my normal connection.... via sudo lsof.

launchd      1                  root  fp.u    systm 0x4026b24b39485ca5       0t0                     [ctl com.apple.net.netagent id 2 unit 17] (fileport=0x2d923)
launchd      1                  root  fp.                                                            unknown file type: 524335 (fileport=0x8002f)
launchd      1                  root  fp.                                                            unknown file type: 539651 (fileport=0x83c03)
launchd      1                  root  fp.u    systm 0x4026b24b394859a5       0t0                     [ctl com.apple.net.netagent id 2 unit 14] (fileport=0x83d03)
launchd      1                  root  fp.u    systm 0x4026b24b394861a5       0t0                     [ctl com.apple.net.netagent id 2 unit 28] (fileport=0x88d63)
launchd      1                  root  fp.u    systm 0x4026b24b39486025       0t0                     [ctl com.apple.net.netagent id 2 unit 25] (fileport=0x8a35f)
launchd      1                  root  fp.u    systm 0x4026b24b39485225       0t0                     [ctl com.apple.net.netagent id 2 unit 23] (fileport=0x8a45f)
launchd      1                  root  fp.u    systm 0x4026b24b39485e25       0t0                     [ctl com.apple.net.netagent id 2 unit 20] (fileport=0x8a523)
launchd      1                  root  fp.u    systm 0x4026b24b39485da5       0t0                     [ctl com.apple.net.netagent id 2 unit 19] (fileport=0x8a65f)
launchd      1                  root  fp.u    systm 0x4026b24b39485c25       0t0                     [ctl com.apple.net.netagent id 2 unit 15] (fileport=0x8fc27)
launchd      1                  root  fp.u    systm 0x4026b24b39484225       0t0                     [ctl com.apple.net.netagent id 2 unit 16] (fileport=0x91b1f)
launchd      1                  root  fp.u    systm 0x4026b24b39485ba5       0t0                     [ctl com.apple.flow-divert id 1 unit 1] (fileport=0x9c65f)
launchd      1                  root  fp.u    systm 0x4026b24b39485d25       0t0                     [ctl com.apple.net.netagent id 2 unit 18] (fileport=0x9c877)
launchd      1                  root  fp.u    systm 0x4026b24b39485ea5       0t0                     [ctl com.apple.net.netagent id 2 unit 21] (fileport=0x9c95f)
launchd      1                  root  fp.u    systm 0x4026b24b39485f25       0t0                     [ctl com.apple.net.netagent id 2 unit 22] (fileport=0x9ca23)
launchd      1                  root  fp.u    systm 0x4026b24b39485fa5       0t0                     [ctl com.apple.net.netagent id 2 unit 24] (fileport=0x9cb23)
launchd      1                  root  fp.u    systm 0x4026b24b394860a5       0t0                     [ctl com.apple.net.netagent id 2 unit 26] (fileport=0x9cc63)
launchd      1                  root  fp.                                                            unknown file type: 675075 (fileport=0xa4d03)
launchd      1                  root  fp.u    systm 0x4026b24b39486125       0t0                     [ctl com.apple.net.netagent id 2 unit 27] (fileport=0xa6eab)
launchd      1                  root    0w      CHR                3,2       0t0                 338 /dev/null
launchd      1                  root    1u      CHR                3,2       0t0                 338 /dev/null
launchd      1                  root    2w      CHR                0,0    0t5905                 336 /dev/console
launchd      1                  root    3u     unix 0x4026b241a0e8eb1d       0t0                     /private//var/run/syslog
launchd      1                  root    4u     unix 0x4026b241a0e8ebe5       0t0                     ->0x4026b241a0e8eb1d
launchd      1                  root    5w      REG               1,13   4371173             1840342 /private/var/log/com.apple.xpc.launchd/launchd.log
launchd      1                  root    6u     unix 0x4026b241a0e8ecad       0t0                     /private/var/run/cupsd
launchd      1                  root    7r      REG               1,13       358              595398 /private/etc/security/audit_control
launchd      1                  root    8u     unix 0x4026b241a0e8ed75       0t0                     /var/rpc/ncalrpc/NETLOGON
launchd      1                  root    9u     unix 0x4026b241a0e8ecad       0t0                     /private/var/run/cupsd
launchd      1                  root   10u     unix 0x4026b241a0e8ee3d       0t0                     /var/run/vpncontrol.sock
launchd      1                  root   11u     unix 0x4026b241a0e912f5       0t0                     /private/tmp/com.apple.launchd.Zex9oyDTj5/Listeners
launchd      1                  root   12u     unix 0x4026b241a0e8ef05       0t0                     /var/run/portmap.socket
launchd      1                  root   13u     unix 0x4026b241a0e8ee3d       0t0                     /var/run/vpncontrol.sock
launchd      1                  root   14u     unix 0x4026b241a0e8efcd       0t0                     /private/var/run/.sim_diagnosticd_socket
launchd      1                  root   15u     unix 0x4026b241a0e8ef05       0t0                     /var/run/portmap.socket
launchd      1                  root   16u     unix 0x4026b241a0e8f095       0t0                     /var/run/systemkeychaincheck.socket
launchd      1                  root   17u     unix 0x4026b241a0e8efcd       0t0                     /private/var/run/.sim_diagnosticd_socket
launchd      1                  root   18u     unix 0x4026b241a0e8f15d       0t0                     /var/run/mDNSResponder
launchd      1                  root   19u     unix 0x4026b241a0e913bd       0t0                     /var/run/filesystemui.socket
launchd      1                  root   20u     unix 0x4026b241a0e8f225       0t0                     /var/rpc/ncacn_np/lsarpc
launchd      1                  root   21u     unix 0x4026b241a0e912f5       0t0                     /private/tmp/com.apple.launchd.Zex9oyDTj5/Listeners
launchd      1                  root   22u     unix 0x4026b241a0e8f2ed       0t0                     /var/rpc/ncalrpc/lsarpc
launchd      1                  root   23u     unix 0x4026b241a0e8f3b5       0t0                     /var/rpc/ncacn_np/mdssvc
launchd      1                  root   24u     unix 0x4026b241a0e8f225       0t0                     /var/rpc/ncacn_np/lsarpc
launchd      1                  root   25u     unix 0x4026b241a0e8f2ed       0t0                     /var/rpc/ncalrpc/lsarpc
launchd      1                  root   26u     unix 0x4026b241a0e8f47d       0t0                     /var/rpc/ncacn_np/wkssvc

and this from running a grep -R 

./System/iOSSupport/System/Library/Frameworks/NetworkExtensioniOSSupport.framework/Versions/A/Resources/Info.plist:		<key>com.apple.networkextension.app-proxy</key>
./System/iOSSupport/System/Library/Frameworks/NetworkExtensioniOSSupport.framework/Versions/A/Resources/Info.plist:					<string>com.apple.networkextension.app-proxy</string>
./System/iOSSupport/System/Library/Frameworks/NetworkExtensioniOSSupport.framework/Versions/A/Resources/Info.plist:				<string>com.apple.networkextension.app-proxy</string>
./System/iOSSupport/System/Library/Frameworks/NetworkExtensioniOSSupport.framework/Versions/A/Resources/Info.plist:		<key>com.apple.networkextension.packet-tunnel</key>
./System/iOSSupport/System/Library/Frameworks/NetworkExtensioniOSSupport.framework/Versions/A/Resources/Info.plist:					<string>com.apple.networkextension.packet-tunnel</string>
./System/iOSSupport/System/Library/Frameworks/NetworkExtensioniOSSupport.framework/Versions/A/Resources/Info.plist:				<string>com.apple.networkextension.packet-tunnel</string>

since the private relay is bundled into iOS as well... it might be that harmonizing Apple OS and hardware (chips) means less development and more congruity. 

Anyone ever lockout these binaries? Or do you know how to find them? See above.

**Reply by robertgendler** [reply]
No.

**Reply by ghost** [reply]
I blocked apple-dns.net and it looks like it also dealt with the some servers for Limit IP address as well. 
<img width="659" alt="Screenshot 2023-08-22 at 6 45 53 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/e1189d7e-3935-401e-9a4e-d6ab3fdac483">

**Comment by ghost** [comment]
I'm whitelisting iCloud.com sites as needed and Limit IP address tracking turned on... so I manually had to turn it off, even though I'm blocking mask.icloud.com and mask-h2.icloud.com. 
<img width="1680" alt="Screenshot 2023-08-22 at 7 44 06 AM" src="https://github.com/usnistgov/macos_security/assets/102542366/5607aa8f-cf60-4dfa-8d0d-0a9e9db252e3">

**Comment by ghost** [comment]
In digging through LSOF I found these paths tied to what seems to be Private Relay/Limit IP Address Tracking: 

com.apple.net.netagent
/usr/libexec/wifip2pd
/Users/vaughnhart/Library/Caches/GeoServices
/System/Library/Frameworks/CoreTelephony.framework/Support/CommCenter
com.apple.net.utun_control
com.apple.flow-divert
/usr/libexec/networkserviceproxy
/usr/libexec/nesessionmanager
/var/run/pppconfd
/var/run/vpncontrol.sock
